api/src/main/java/edu/wpi/first/shuffleboard/api/sources/DummySource.java

```diff
@@ -29,21 +29,26 @@ public SourceType getType() {
    * If no example source value could be found, then None is returned instead.
    */
   @SuppressWarnings("unchecked")
-  public static Optional<DummySource> forTypes(Set<DataType> types) {
+  public static DataSource<?> forTypes(Set<DataType> types) {
     if (types.isEmpty()) {
-      return Optional.empty();
+      return DataSource.none();
     }
+
     if (types.stream().anyMatch(DataType::isComplex)) {
       DataType type = types.stream().filter(DataType::isComplex).findFirst().get();
-      return Optional.of(new DummySource(type, type.getDefaultValue()));
+      return new DummySource(type, type.getDefaultValue());
     } else {
       DataType type = (DataType) types.toArray()[0];
-      return Optional.of(new DummySource(type, type.getDefaultValue()));
+      return new DummySource(type, type.getDefaultValue());
     }
   }
 
-  public static Optional<DummySource> forTypes(DataType... types) {
+  public static DataSource<?> forTypes(DataType... types) {
     return forTypes(ImmutableSet.copyOf(types));
   }
 
+  public static DataSource forName(String name) {
+    return DummySource.forTypes(DataTypes.getDefault().forName(name)
```

Comment:
This could be replaced with

```java
return DataTypes.getDefault().forName(name)
    .map(DummySource::forTypes)
    .orElseThrow(() -> new IllegalArgumentException("No DataType " + name));
```

Reply:
seems clearer this way?